syntax = "proto3";

message Empty {
}

message AiddlState {
  string state = 1;
}

message AiddlGoal {
  string goal = 1;
}

message AiddlResult {
  string result = 1;
}

service MazeGUI {
  rpc requestTask(Empty) returns(AiddlGoal);
  rpc processTaskResult(AiddlResult) returns(Empty);
  rpc getState(Empty) returns(AiddlState);
  rpc visualizeState(AiddlState) returns(Empty);
}
